### PR TITLE
feat: ruby multipart

### PR DESCRIPTION
**Pull request by loks0n**

### CODE REVIEW

File: templates/ruby/lib/container/payload.rb.twig [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        mime_type
+      end
+    end
+
+    def self.from_path(path, filename: nil, mime_type: nil)
+      raise "File not found" unless File.exist?(path)
+      new(nil, path, filename, mime_type)
+    end
+
+    def self.from_data(data, filename: nil, mime_type: nil)
```
Comment (christyjacob4) [inline templates/ruby/lib/container/payload.rb.twig:37]:
```suggestion
    def self.from_binary(data, filename: nil, mime_type: nil)
```

File: templates/ruby/lib/container/payload.rb.twig [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+    def to_binary
+      @data || File.binread(@path)
+    end
+
+    def to_json
+      JSON.parse(@data)
+    end
+
+    def to_s
```
Comment (christyjacob4) [inline templates/ruby/lib/container/payload.rb.twig:57]: is it not possible to define a to_string? it it a reserved function ?

File: templates/ruby/lib/container/multipart.rb.twig [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+                else
+                    begin
+                        h[name] = part[:contents].force_encoding("utf-8")
+                    rescue
+                        h[name] = part[:contents]
+                    end
+                end
+            end
+
+            h
```
Comment (Meldiron) [inline templates/ruby/lib/container/multipart.rb.twig:117]: h? 😅

File: templates/ruby/lib/container/payload.rb.twig [hunk 4]
```diff
@@ -0,0 +1,77 @@
+module Appwrite
+  class Payload
+    attr_reader :data
+    attr_reader :filename
+
+    def initialize(data: nil, path: nil, filename: nil)
```
Comment (Meldiron) [inline templates/ruby/lib/container/payload.rb.twig:6]: Do we need path here?

File: templates/ruby/lib/container/payload.rb.twig [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+      @path = path
+      @data = data
+      @filename = filename
+
+      @size = if @data then
+        @data.bytesize
+      else
+        File.size(@path)
```
Comment (Meldiron) [inline templates/ruby/lib/container/payload.rb.twig:14]: Do we need this size?

File: templates/ruby/lib/container/payload.rb.twig [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+      end
+    end
+
+    def read(offset: 0, length: nil)
```
Comment (Meldiron) [inline templates/ruby/lib/container/payload.rb.twig:18]: Having generic read seems off. Would read work with strings or jsons? Is this needed specifically for files?

File: templates/ruby/lib/container/payload.rb.twig [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    # @param [String] path
+    # @param [String, nil] filename
+    # @return [Payload]
+    def self.from_file(path, filename: nil)
+        filename = if filename.nil? then
+          File.basename(path)
+        else
+          filename
+        end
+      new(nil, path, filename)
```
Comment (Meldiron) [inline templates/ruby/lib/container/payload.rb.twig:35]: Hmm, data nil and we pass file?
Can we instead simply get data here, and pass it? That way constructor would probably not need path anymore.

File: templates/ruby/lib/container/payload.rb.twig [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    end
+
+    # @param [String] b
+    # @param [String, nil] filename
+    def self.from_binary(b, filename: nil)
```
Comment (Meldiron) [inline templates/ruby/lib/container/payload.rb.twig:40]: lets do `bytes` instead of `b`

File: templates/ruby/lib/container/payload.rb.twig (hunk 4 above)
Comment (loks0n) [reply templates/ruby/lib/container/payload.rb.twig:6]: It is stored to enable loading the file in chunks

File: templates/ruby/lib/container/payload.rb.twig (hunk 5 above)
Comment (loks0n) [reply templates/ruby/lib/container/payload.rb.twig:14]: It is stored to enable loading the file in chunks

File: templates/ruby/lib/container/payload.rb.twig (hunk 6 above)
Comment (loks0n) [reply templates/ruby/lib/container/payload.rb.twig:18]: It's need for the chunking logic in the client

File: templates/ruby/lib/container/payload.rb.twig (hunk 7 above)
Comment (loks0n) [reply templates/ruby/lib/container/payload.rb.twig:35]: In the case of large files, you load everything into memory this way.

We can do it, but it's not how it was before

File: templates/ruby/lib/container/payload.rb.twig (hunk 2 above)
Comment (loks0n) [reply templates/ruby/lib/container/payload.rb.twig]: to_string is not reserved, but to_s is idiomatic 

I've gone with both

**Review by christyjacob4 — APPROVED** [review]

**Pull request merged**